test/unittests/oom_test.cpp

```diff
@@ -0,0 +1,92 @@
+// Fizzy: A fast WebAssembly interpreter
+// Copyright 2021 The Fizzy Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "execute.hpp"
+#include "parser.hpp"
+#include <gtest/gtest.h>
+#include <test/utils/asserts.hpp>
+#include <test/utils/execute_helpers.hpp>
+#include <test/utils/hex.hpp>
+
+#ifdef __linux__
+#include <sys/resource.h>
+#endif
+
+#ifndef __has_feature
+#define __has_feature(X) 0
+#endif
+
+#if defined(__SANITIZE_ADDRESS__) || __has_feature(address_sanitizer)
+#define ASAN 1
+#else
+#define ASAN 0
+#endif
+
+namespace
+{
+constexpr uint32_t OSMemoryLimit = 1024 * 1024;
+
+bool try_set_memory_limit(size_t size) noexcept;
+bool restore_memory_limit() noexcept;
+
+#if !ASAN && defined(__linux__) && (defined(__x86_64__) || defined(__i386__))
+
+rlimit orig_limit;
+
+bool try_set_memory_limit(size_t size) noexcept
+{
+    [[maybe_unused]] int err;
+    err = getrlimit(RLIMIT_AS, &orig_limit);
```

Comment:
We could also have some stack tests using `RLIMIT_STACK`, i.e. having a known limit.